client/src/components/ElementEditor/Element.js

```diff
@@ -215,6 +296,81 @@ class Element extends Component {
     }
   }
 
+  handleSaveButtonClick() {
+    this.setState({
+      ensureFormRendered: true,
+      doSaveElement: true,
+    });
+  }
+
+  handlePublishButtonClick() {
+    this.setState({
+      justClickedPublishButton: true,
+      ensureFormRendered: true,
+    });
+    if (this.props.formDirty) {
+      // Save the element first before publishing, which may trigger validation errors
+      this.props.submitForm();
+    } else {
+      // Just publish the element straight away without saving first
+      this.setState({
+        doPublishElement: true,
+      });
+    }
+  }
+
+  handleAfterSave() {
+    this.setState({
+      doSaveElement: false,
+    });
+  }
```

Comment:
Have added comment on https://github.com/silverstripe/silverstripe-elemental/pull/1150/files#r1527640688

tl;dr it's all because we're doing a remote submit using redux-form which doesn't return a promise

Reply:
That doesn't explain why the toast can't be here? Is this not after the save event has occurred?

Reply:
Don't bother answering that - instead just refer to https://github.com/silverstripe/silverstripe-elemental/pull/1150#discussion_r1531092733.